designs/local-pv/lvm/snapshot-restore.md

```diff
@@ -0,0 +1,163 @@
+---
+oep-number: OEP 4071
+title: Add Snapshot Restore Feature to OpenEBS LVM
+authors:
+  - "@rohan2794"
+owners:
+  - "@rohan2794"
+editor: TBD
+creation-date: 03/10/2025
+last-updated: 03/10/2025
+status: provisional
+---
+
+# Add Snapshot Restore Feature to OpenEBS LVM
+> Warning: Restoring from a snapshot is available exclusively for thinly provisioned volume snapshots.
+
+## Table of Contents
+
+1. [Overview](#overview)
+2. [Motivation](#motivation)
+3. [Goals](#goals)
+4. [Non-Goals](#non-goals)
+5. [Proposal](#proposal)
+6. [User Stories](#user-stories)
+7. [Implementation Details](#implementation-details)
+8. [Testing](#testing)
+
+## Overview
+
+This OEP proposes adding Snapshot Restore feature to OpenEBS LVM.
+This feature would enable users to create restore from snapshot of thinly provisioned volume.
+
+## Motivation
+
+OpenEBS LVM supports snapshots, but restoring from them is not supported. This limitation creates challenges when attempting to recover data from snapshots.
+
+## Goals
+
+- Enable users to create restore from snapshot of thinly provisioned volume.
+
+## Non-Goals
+
+- Including support for restoring from snapshots of thick-provisioned volumes as part of this OEP.
+
+## Proposal
+
+For restoring from snapshot of thin-provisioned volume, a new thin-provisioned snapshot is created from the existing snapshot. Since LVM snapshots are writable, these can be exposed as new PVCs without losing the original snapshot’s identity. This allows multiple restores from the same snapshot, resulting in multiple independent copies. The restored thin snapshot volumes are activated as Read-Write and presented to Kubernetes as PersistentVolume objects.
```

Comment:
```suggestion
For restoring from snapshot of thin-provisioned volume, a new thin-provisioned snapshot is created from the existing snapshot. Since LVM snapshots are writable, these can be exposed as new PVCs without losing the original snapshot’s identity. This allows multiple restores from the same snapshot, resulting in multiple independent copies. The restored thin snapshot volume is activated as Read-Write and presented to Kubernetes as PersistentVolume object.
```